ACPs/99-validatorsetmanager-contract/README.md

```diff
@@ -394,36 +291,57 @@ graph LR
 
-A work in progress reference implementation is available in the [Suzaku Contracts Library](https://github.com/suzaku-network/suzaku-contracts-library/blob/acp-99-implementation/README.md#acp99-contracts-library) repository. It will be updated until this ACP is considered `Implementable` based on the outcome of the discussion.
+```mermaid
+---
+title: ACP-99 Single-Contract Architecture implementing PoA
+---
+  classDiagram
+  class ACP99Manager {
+    initializeValidatorSet
+    initiateValidatorRegistration
+    completeValidatorRegistration
+    initiateValidatorWeightUpdate
+    completeValidatorWeightUpdate
+  }
+  <<abstract>> ACP99Manager
+
+  class ValidatorManager {
+    completeValidatorRegistration
+  }
+  <<abstract>> ValidatorManager
+  class PoAValidatorManager {
+    initiateValidatorRegistration
+    initiateEndValidation
+    completeEndValidation
+  }
+
+  ACP99Manager <|--ValidatorManager
+  ValidatorManager <|-- PoAValidatorManager
+```
+
+This implementation is available in Ava Labs' [ICM Contracts Repository](https://github.com/ava-labs/icm-contracts/tree/main/contracts/validator-manager).
 
 ## Security Considerations
 
-The audit process of the `ACP99Manager` contract is of the utmost importance for the future of the Avalanche ecosystem as most Subnets would rely upon it to secure their Subnet.
+The audit process of `ACP99Manager` and reference implementations is of the utmost importance for the future of the Avalanche ecosystem as most L1s would rely upon it to secure their L1.
 
 ## Open Questions
 
 ### Is there an interest to keep historical information about the validator set on the manager chain?
 
-The functions `getValidation` and `getValidatorValidations` would allow to retrieve historical information about the validator set directly from state, notably each validator's performance (uptime) during past validations.
-If we don't keep track of the historical information in the `ACP99Manager` contract, this information will still be available in archive nodes and offchain tools (e.g. explorers).
-
-### Is an upgradeable architecture more appropriate?
-
-The Teleporter team has been working on its own implementation of a `ValidatorManager` contract in the [teleporter](https://github.com/ava-labs/teleporter/blob/main/contracts/validator-manager/ValidatorManager.sol) repository that is based on abstract and upgradeable contracts.
-
-### Should the `ACP99Manager` include a churn control mechanism?
+It's undefined if `getValidation` should return allow historical information about ended validations. Should `ACP99Manager` enforce that validation information be kept in the contract's state indefinitely? Note that validator performance (uptime) is _not_ specified in the `ACP99Manager` interface, as it may not be relevant to some applications (e.g. PoA). Historical uptime would be a useful metric to query, but it may be more appropriately left to implementations to enforce.
```

Comment:
```suggestion
It's undefined if `getValidation` should return historical information about ended validations. Should `ACP99Manager` enforce that validation information be kept in the contract's state indefinitely? Note that validator performance (uptime) is _not_ specified in the `ACP99Manager` interface, as it may not be relevant to some applications (e.g. PoA). Historical uptime would be a useful metric to query, but it may be more appropriately left to implementations to enforce.
```